### PR TITLE
[SYCL] [L0] Remove unneeded backwards compatibility of 2023.2 make_queue and get_native.

### PR DESCRIPTION
**Pull request by rdeodhar**
The 2023.2 compiler release changed the implementation of get_native(queue) and make_queue to allow use of immediate command lists while maintaining binary compatibility with earlier compiler releases. In 2024.0 we no longer need backwards compatibility and this change removes it.

### CODE REVIEW

File: sycl/include/sycl/detail/pi.h [hunk 1]
```diff
@@ -83,9 +83,9 @@
 // 12.25 Added PI_EXT_DEVICE_INFO_ATOMIC_FENCE_ORDER_CAPABILITIES and
 // PI_EXT_DEVICE_INFO_ATOMIC_FENCE_SCOPE_CAPABILITIES for piDeviceGetInfo.
 // 12.26 Added piextEnqueueReadHostPipe and piextEnqueueWriteHostPipe functions.
-// 12.27 Added new queue create and get APIs for immediate commandlists
-// piextQueueCreate2, piextQueueCreateWithNativeHandle2,
-// piextQueueGetNativeHandle2
+// 12.27 Added properties parameter to piextQueueCreateWithNativeHandle and
+// changed native handle type of piextQueueCreateWithNativeHandle and
+// piextQueueGetNativeHandle
```
Comment (smaslov-intel) [inline sycl/include/sycl/detail/pi.h:88]: Please make it a new breaking change instead, i.e. 13.31

File: sycl/plugins/cuda/pi_cuda.cpp
Comment (smaslov-intel) [inline sycl/plugins/cuda/pi_cuda.cpp:1]: why are so many unrelated changes to this file?

File: sycl/plugins/hip/pi_hip.cpp [hunk 2]
```diff
@@ -2692,20 +2692,15 @@ pi_result hip_piQueueFlush(pi_queue command_queue) {
 ///
 /// \return PI_SUCCESS
 pi_result hip_piextQueueGetNativeHandle(pi_queue queue,
-                                        pi_native_handle *nativeHandle) {
+                                        pi_native_handle *nativeHandle,
+                                        int32_t *NativeHandleDesc) {
+  (void)NativeHandleDesc;
```
Comment (smaslov-intel) [inline sycl/plugins/hip/pi_hip.cpp:2697]: I think it is worth initializing desc with 0

File: sycl/plugins/opencl/pi_opencl.cpp [hunk 3]
```diff
@@ -2245,14 +2238,9 @@ pi_result piextContextGetNativeHandle(pi_context context,
...
-  return piextGetNativeHandle(queue, nativeHandle);
-}
-
-pi_result piextQueueGetNativeHandle2(pi_queue queue,
-                                     pi_native_handle *nativeHandle,
-                                     int32_t *NativeHandleDesc) {
-  (void)NativeHandleDesc;
+                                    pi_native_handle *nativeHandle,
+                                    int32_t *nativeHandleDesc) {
+  (void)nativeHandleDesc;
```
Comment (smaslov-intel) [inline sycl/plugins/opencl/pi_opencl.cpp:2243]: please initialize to 0

**Review by smaslov-intel — COMMENTED** [review]
LGTM, just few minor comments

File: sycl/include/sycl/detail/pi.h (hunk 1 above)
Comment (rdeodhar) [reply sycl/include/sycl/detail/pi.h:88]: OK, done.

File: sycl/plugins/cuda/pi_cuda.cpp
Comment (rdeodhar) [reply sycl/plugins/cuda/pi_cuda.cpp:1]: Merge problem, now fixed.

File: sycl/plugins/hip/pi_hip.cpp (hunk 2 above)
Comment (rdeodhar) [reply sycl/plugins/hip/pi_hip.cpp]: OK, done.

File: sycl/plugins/opencl/pi_opencl.cpp (hunk 3 above)
Comment (rdeodhar) [reply sycl/plugins/opencl/pi_opencl.cpp]: Done.

**Review by smaslov-intel — APPROVED** [review]

**Review by jandres742 — APPROVED** [review]

File: sycl/plugins/cuda/pi_cuda.cpp [hunk 4]
```diff
@@ -2773,41 +2773,39 @@ pi_result cuda_piQueueFlush(pi_queue command_queue) {
...
 ///
 /// \param[in] queue The PI queue to get the native CUDA object of.
+/// \param[in] NativeHandleDesc Pointer to additional native handle info.
 /// \param[out] nativeHandle Set to the native handle of the PI queue object.
 ///
 /// \return PI_SUCCESS
 pi_result cuda_piextQueueGetNativeHandle(pi_queue queue,
-                                         pi_native_handle *nativeHandle) {
+                                         pi_native_handle *nativeHandle,
+                                         int32_t *NativeHandleDesc) {
```
Comment (jchlanda) [inline sycl/plugins/cuda/pi_cuda.cpp:2782]: Nit: variable names seem to be following camelCase starting with small char (same for pi_hip).

File: sycl/plugins/cuda/pi_cuda.cpp [hunk 5]
```diff
@@ -2773,41 +2773,39 @@ pi_result cuda_piQueueFlush(pi_queue command_queue) {
...
 /// \param ownNativeHandle tells if SYCL RT should assume the ownership of
 ///        the native handle, if it can.
 ///
 /// \return TBD
-pi_result cuda_piextQueueCreateWithNativeHandle(pi_native_handle nativeHandle,
-                                                pi_context context,
-                                                pi_device device,
-                                                bool ownNativeHandle,
-                                                pi_queue *queue) {
+pi_result cuda_piextQueueCreateWithNativeHandle(
```
Comment (jchlanda) [inline sycl/plugins/cuda/pi_cuda.cpp:2801]: Nit: camelCase, as above.

**Review by jchlanda — APPROVED** [review]
CUDA and HIP 👍

File: sycl/plugins/cuda/pi_cuda.cpp (hunk 5 above)
Comment (rdeodhar) [reply sycl/plugins/cuda/pi_cuda.cpp:2801]: The naming convention for parameters is at least consistent with other get_native and make APIs.

**Review by gmlueck — APPROVED** [review]
spec changes are OK.  (Just an addition of a newline character)

**Review by v-klochkov — APPROVED** [review]
